src/sage/crypto/lwe.py

```diff
@@ -298,6 +298,7 @@ def __init__(self, n, q, D, secret_dist='uniform', m=None):
 
             sage: from numpy import std
             sage: while abs(std([e if e <= 200 else e-401 for e in S()]) - 3.0) > 0.01:
+            ....:     L = []  # reset L to avoid quadratic behaviour
```

Comment:
isn't the idea of this test that by increasing the number of samples, the error bound will be hit?

Reply:
I'm not sure. To be honest I'm not sure what is the role of this test, but the previous implementation exhibits quadratic behavior which is the cause of this test usually being ok but sometimes being very slow:
```
sage -t --warn-long --random-seed=110988274722243807127083377606682083581 src/sage/crypto/lwe.py
**********************************************************************
File "src/sage/crypto/lwe.py", line 300, in sage.crypto.lwe.LWE.__init__
Warning, slow doctest:
    while abs(std([e if e <= 200 else e-401 for e in S()]) - 3.0) > 0.01:
        add_samples()
Test ran for 16.66 s, check ran for 0.00 s
    [112 tests, 17.29 s]
```
vs.
```
sage -t --warn-long 0.4 --random-seed=1 src/sage/crypto/lwe.py
**********************************************************************
File "src/sage/crypto/lwe.py", line 300, in sage.crypto.lwe.LWE.__init__
Warning, slow doctest:
    while abs(std([e if e <= 200 else e-401 for e in S()]) - 3.0) > 0.01:
        add_samples()
Test ran for 0.47 s, check ran for 0.00 s
    [112 tests, 1.37 s]
```

As far as I understand, they want to show that these samples indeed have a normal distribution with standard deviation 3.0.  They take 1000 samples and want the standard deviation of these to be close to 3.0. Otherwise they keep adding samples, etc. until the standard deviation of the samples is indeed close to 3.0.

However,  the way this is implemented it becomes O(n^2) when they have to try 1000n samples.

With my change, instead of adding more samples, we take a new set of 1000 samples. In this way, trying 1000n samples is O(n). So, even if we have to try more times, this is better.

Moreover, now this is linear instead of quadratic, it's even faster too try a sample of 100.

Summary: the new way the test is done, it keeps computing the standard deviation of 100 samples until it's really close to 3.0.

Reply:
Let me know if you are happy with my explanation. Otherwise, I'll revert and place a `# long time` label (although I'd be more inclined to just nuke the test).

I think this was the only non-cosmetic objection you had (and the cosmetic ones are more or less all addressed).

Reply:
I think I would be more comfortable if we just get rid of the `while` loop, compute and print the std and mark the result as random

Reply:
That said, your solution is of course fine.